### ISSUE TITLE
JSON-LD serialization fails, but turtle works

### ISSUE THREAD

**Issue by apirogov** [oxigraph/oxrdflib#42]
Minimal working example:

```python
from rdflib import Graph
import json

test_data = [{
    "@context": "https://schema.org/",
    "@id": "https://doi.org/some/doi",
    "@type": "Dataset",
    "creator": [{"@type": "Person", "name": "Some person"}],
}]

g = Graph(store="Oxigraph").parse(data=json.dumps(test_data), format='json-ld')
print(g.serialize())  # works
print(g.serialize(format='json-ld'))  # error
```

The JSON-LD serialization causes an exception:
```
Traceback (most recent call last):
  File "/local/home/a.pirogov/projects/data_harvesting/scratch.py", line 13, in <module>
    print(g.serialize(format='json-ld'))  # error
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/graph.py", line 1337, in serialize
    serializer.serialize(stream, base=base, encoding="utf-8", **args)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 84, in serialize
    obj = from_rdf(
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 131, in from_rdf
    result = converter.convert(graph)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 172, in convert
    nodes = self.from_graph(g)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 204, in from_graph
    self.process_subject(graph, s, nodemap)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 225, in process_subject
    self.add_to_node(graph, s, p, o, node, nodemap)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 291, in add_to_node
    node = self.to_raw_value(graph, s, o, nodemap)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 332, in to_raw_value
    onode = self.process_subject(graph, o, nodemap)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 225, in process_subject
    self.add_to_node(graph, s, p, o, node, nodemap)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 291, in add_to_node
    node = self.to_raw_value(graph, s, o, nodemap)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 321, in to_raw_value
    coll = self.to_collection(graph, o)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/plugins/serializers/jsonld.py", line 367, in to_collection
    if l_ != RDF.nil and not graph.value(l_, RDF.first):
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/graph.py", line 1039, in value
    retval = next(values)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/graph.py", line 859, in objects
    for s, p, o in self.triples((subject, predicate, None)):
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/rdflib/graph.py", line 594, in triples
    for (_s, _p, _o), cg in self.__store.triples((s, p, o), context=self):
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/data-harvesting-Lq057YRM-py3.9/lib/python3.9/site-packages/oxrdflib/__init__.py", line 94, in triples
    for q in self._inner.quads_for_pattern(*_to_ox_quad_pattern(triple_pattern, context))
TypeError: Literal is not an RDF named or blank node
```

**Comment by Tpt** [comment]
Thank you for this great bug report! It seems that the JSON-LD serializer tries to get triples for some invalid triple patterns like ones with `subject = Literal("Some person")`, making pyoxigraph raise an exception. #43 will catch the exception and return an empty iterator in this case

**Comment by Tpt** [comment]
Should be fixed in v0.3.7
